server/api/hook.go

```diff
@@ -81,112 +81,122 @@ func PostHook(c *gin.Context) {
 
 	tmpRepo, build, err := _remote.Hook(c.Request)
 	if err != nil {
-		log.Error().Msgf("failure to parse hook. %s", err)
-		_ = c.AbortWithError(400, err)
+		msg := "failure to parse hook"
+		log.Debug().Err(err).Msg(msg)
+		c.String(http.StatusBadRequest, "%s: %v", msg, err)
 		return
 	}
 	if build == nil {
-		c.Writer.WriteHeader(200)
+		c.String(http.StatusOK, "ignoring hook: hook parse result in empty build")
 		return
 	}
 	if tmpRepo == nil {
-		log.Error().Msgf("failure to ascertain repo from hook.")
-		c.Writer.WriteHeader(400)
+		c.String(http.StatusBadRequest, "failure to ascertain repo from hook")
 		return
 	}
 
 	// skip the build if any case-insensitive combination of the words "skip" and "ci"
 	// wrapped in square brackets appear in the commit message
 	skipMatch := skipRe.FindString(build.Message)
 	if len(skipMatch) > 0 {
-		log.Info().Msgf("ignoring hook. %s found in %s", skipMatch, build.Commit)
-		c.Writer.WriteHeader(204)
+		msg := fmt.Sprintf("ignoring hook: %s found in %s", skipMatch, build.Commit)
+		log.Debug().Msg(msg)
+		c.String(http.StatusNoContent, msg)
 		return
 	}
 
 	repo, err := _store.GetRepoName(tmpRepo.Owner + "/" + tmpRepo.Name)
 	if err != nil {
-		log.Error().Msgf("failure to find repo %s/%s from hook. %s", tmpRepo.Owner, tmpRepo.Name, err)
-		_ = c.AbortWithError(404, err)
+		msg := fmt.Sprintf("failure to get repo %s/%s from store", tmpRepo.Owner, tmpRepo.Name)
+		log.Error().Err(err).Msg(msg)
+		c.String(http.StatusNotFound, "%s: %v", msg, err)
 		return
 	}
 	if !repo.IsActive {
-		log.Error().Msgf("ignoring hook. %s/%s is inactive.", tmpRepo.Owner, tmpRepo.Name)
```

Comment:
Why did you removed the error logging here?

Reply:
since this is no server error just wrong user config - that can be debugged by response of hook request at forges